File: src/main/java/com/fifo/ticketing/domain/book/repository/BookScheduleRepository.java
package com.fifo.ticketing.domain.book.repository;

import com.fifo.ticketing.domain.book.entity.BookScheduledTask;
import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

@Repository
public interface BookScheduleRepository extends JpaRepository<BookScheduledTask, Long> {

    @Query("SELECT s FROM BookScheduledTask s WHERE s.taskStatus = 'PENDING'")
    List<BookScheduledTask> findAllPendingTasks();

}

File: src/main/java/com/fifo/ticketing/domain/book/repository/BookScheduleRepository.kt
package com.fifo.ticketing.domain.book.repository

import com.fifo.ticketing.domain.book.entity.BookScheduledTask
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository

@Repository
interface BookScheduleRepository : JpaRepository<BookScheduledTask, Long> {

    @Query("SELECT s FROM BookScheduledTask s WHERE s.taskStatus = 'PENDING'")
    fun findAllPendingTasks(): List<BookScheduledTask>
}
